lib/compress/zstdmt_compress.c

```diff
@@ -528,17 +548,18 @@ static unsigned computeNbChunks(size_t srcSize, unsigned windowLog, unsigned nbT
     return (multiplier>1) ? nbChunksLarge : nbChunksSmall;
 }
 
-
-size_t ZSTDMT_compress_advanced(ZSTDMT_CCtx* mtctx,
-                               void* dst, size_t dstCapacity,
-                         const void* src, size_t srcSize,
-                         const ZSTD_CDict* cdict,
-                               ZSTD_parameters const params,
-                               unsigned overlapLog)
+static size_t ZSTDMT_compress_advanced_internal(
+                ZSTDMT_CCtx* mtctx,
+                void* dst, size_t dstCapacity,
+          const void* src, size_t srcSize,
+          const ZSTD_CDict* cdict,
+                ZSTD_CCtx_params const cctxParams,
+                unsigned overlapLog)
```

Comment:
I think we could go "all in", and use `cctxParams` to transmit _all_ parameters.
This includes `overlapLog`, and also `nbThreads` and `jobSize`.
Prototype and body should be adapted to reflect it.